src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -792,17 +795,19 @@ export class ResultsViewPageStore {
     readonly cnSegmentsByChromosome = remoteData<{ [chromosome: string]: MobxPromise<CopyNumberSeg[]> }>({
         await: () => [
             this.genes,
-            this.samples
+            this.samples,
+            this.referenceGenes
         ],
         invoke: () => {
-            if (this.genes.result) {
-                return Promise.resolve(_.reduce(_.uniq(this.genes.result.map(g => g.chromosome)), (map: { [chromosome: string]: MobxPromise<CopyNumberSeg[]> }, chromosome: string) => {
-                    map[chromosome] = remoteData<CopyNumberSeg[]> ({
-                        invoke: () => fetchCopyNumberSegmentsForSamples(this.samples.result, chromosome)
-                    });
-
-                    return map;
-                }, {}));
+            if (this.referenceGenes.result) {
```

Comment:
if you're awaiting referenceGenes, then `result` will always be there. no need to check